### PR TITLE
Cleanup add_expand_vars

### PR DESCRIPTION
**Pull request by dodecatheon**
Cleaned up code.

### CODE REVIEW

File: lib/ramble/ramble/application.py [hunk 1]
```diff
@@ -519,6 +517,8 @@ def add_expand_vars(self, workspace):
                 if var not in self.variables.keys():
                     self.variables[var] = conf['default']
 
+    def _inject_chained_commands(self, executables):
```
Comment (douglasjacobsen) [inline lib/ramble/ramble/application.py:520]: I think this should be called `_inject_commands`. It just builds out all of the commands for the experiment, not just chained commands (though it injects those too).

**Review by douglasjacobsen — CHANGES_REQUESTED** [review]

File: lib/ramble/ramble/application.py (hunk 1 above)
Comment (dodecatheon) [reply lib/ramble/ramble/application.py]: Done

File: lib/ramble/ramble/application.py [hunk 2]
```diff
@@ -37,6 +37,10 @@
 from ramble.language.application_language import ApplicationMeta, register_builtin
 from ramble.error import RambleError
 
+# # For checking whether an object is iterable before iterating over it.
```
Comment (douglasjacobsen) [inline lib/ramble/ramble/application.py:40]: These should be removed.

File: lib/ramble/ramble/application.py [hunk 3]
```diff
@@ -604,6 +605,31 @@ def add_expand_vars(self, workspace):
...
 
+    def add_expand_vars(self, workspace):
+        """Add application specific expansion variables
+
+        Applications require several variables to be defined to function properly.
+        This method defines these variables, including:
+        - command: set to the commands needed to execute the experiment
+        - spack_setup: set to an empty string, so spack applications can override this
+        """
+        if self._vars_are_expanded:
```
Comment (douglasjacobsen) [inline lib/ramble/ramble/application.py:616]: Is this still used? I don't see it being set in any of the add expand vars functions.

File: lib/ramble/ramble/test/application_tests.py [hunk 4]
```diff
@@ -299,3 +305,291 @@ def test_register_builtin_app(mutable_mock_repo):
...
+    executables, inputs = executable_application_instance._get_executables_and_inputs()
+
+    executable_application_instance._set_input_path(inputs)
+
+    executable_application_instance.workload_variables = {'test_wl2': {'n_ranks':
+                                                                       {'default': '1'}}}
+
+    executable_application_instance._set_default_experiment_variables()
+
+    # print("\n\nexecutables:", executables, "\n\n")
```
Comment (douglasjacobsen) [inline lib/ramble/ramble/test/application_tests.py:512]: Probably don't need these either

File: lib/ramble/ramble/test/application_tests.py [hunk 5]
```diff
@@ -299,3 +305,291 @@ def test_register_builtin_app(mutable_mock_repo):
...
+    # print("\n\ninstance.nexecutables:", executable_application_instance.executables, "\n\n")
+    executable_application_instance.chain_prepend = []
+    executable_application_instance._inject_commands(executables)
+    # print("\n\ncommand:", executable_application_instance.variables['command'], "\n\n")
```
Comment (douglasjacobsen) [inline lib/ramble/ramble/test/application_tests.py:516]: Or this

**Review by douglasjacobsen — CHANGES_REQUESTED** [review]
Thanks, just a few changes remaining

File: lib/ramble/ramble/application.py (hunk 2 above)
Comment (dodecatheon) [reply lib/ramble/ramble/application.py:40]: done

File: lib/ramble/ramble/test/application_tests.py (hunk 4 above)
Comment (dodecatheon) [reply lib/ramble/ramble/test/application_tests.py:512]: done

**Review by douglasjacobsen — APPROVED** [review]

**Pull request merged**